From 6a0ff9efe948473244122365e1d97b8a6d0d8302 Mon Sep 17 00:00:00 2001
From: Pavel Hrdina <phrdina@redhat.com>
Date: Tue, 30 Jun 2015 10:21:43 +0200
Subject: [PATCH] virt-viewer: set keepAlive on libvirt connection

Setting the keepAlive on libvirt connection is needed in order to
receive the CloseCallback event.

Resolves: rhbz#1164052
(cherry picked from commit 08378ec4dc3623792c64a3bae6279eac1c3c153e)
---
 src/virt-viewer.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/virt-viewer.c b/src/virt-viewer.c
index c1768b3..aefa690 100644
--- a/src/virt-viewer.c
+++ b/src/virt-viewer.c
@@ -966,6 +966,10 @@ virt_viewer_connect(VirtViewerApp *app)
         g_debug("Unable to register close callback on libvirt connection");
     }
 
+    if (virConnectSetKeepAlive(priv->conn, 5, 3) < 0) {
+        g_debug("Unable to set keep alive");
+    }
+
     return 0;
 }
 
